pages/AdListPage/[id]/index.js

```diff
@@ -0,0 +1,122 @@
+import Image from "next/image";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+import { fetcher } from "../../../helpers/api";
+import styled from "styled-components";
+
+export default function AdDetailsPage() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const { data: ad, error } = useSWR(`/api/ads/${id}`, fetcher);
+
+  if (error) return <h1>...sorry cannot load ads</h1>;
+
+  if (!ad) return <h1>...please wait while loading...</h1>;
+  console.log(ad);
+
+  return (
+    <StyledArticle>
+      <UserContainer>
+        <UserProfilPhoto
+          src={
+            ad.userPictureSrc == ""
+              ? `https://source.unsplash.com/random/?person ${ad.userName}`
+              : ad.userPictureSrc
+          }
+          width={40}
+          height={40}
+          alt={`Profilphoto of ${ad.userName}`}
+        ></UserProfilPhoto>
+        <h2>{ad.userName}´s Ad</h2>
+      </UserContainer>{" "}
+      <ImageContainer>
+        <Image
+          src={
+            ad.adPictureSrc == ""
+              ? `https://source.unsplash.com/random/1920x1080/?${ad.tags}`
+              : ad.adPictureSrc
+          }
+          fill
+          alt={`Examplephoto of ${ad.adTitle}`}
+        ></Image>
+      </ImageContainer>
+      <h3>{ad.adTitle}</h3>
+      <Description>
+        <p>Task description:</p>
+        {ad.adDescription}
+      </Description>
+      <Attributes>
+        <p>category: {ad.category}</p>
+        <p>
+          Tags:
+          {ad.tags.map((tag) => {
+            return <span> {tag} </span>;
+          })}{" "}
+        </p>
```

Comment:
I'm pretty sure the console was screaming at you that the `key prop` is missing. 😁
The tags list screams for an `<ul>` with `<li>`  🤬
```suggestion
        <ul>
          Tags:
          {ad.tags.map((tag, index) => {
            return <li key={index}   > {tag} </li>;
          })}
        </ul>
```